ennMethods.rb

```diff
@@ -0,0 +1,101 @@
+# Module
+module Enumerable
+  #my_each: clone of each
+  def my_each
+    for x in self do
+      yield(x)
+    end
+    return self
+  end
+
+  #my_each_with_index: clone of each.with_index
+  def my_each_with_index
+    for i in 0...self.length do
+      yield(self[i], i)
+    end
+    return self
+  end
+
+  #my_select: clone of select
+  def my_select
+    i = 0
+    result = []
+    self.my_each do |x|
+      if yield(x)
+        result[i] = x
+        i += 1
+      end
+    end
+    return result
+  end
+
+  #my_all?: clone of all?
+  def my_all?
+    test = true
+    self.my_each do |x|
+      test = test && yield(x)
+    end
+    return test
+  end
+
+  #my_any?: clone of any?
+  def my_any?
+    test = false
+    self.my_each do |x|
+      test = test || yield(x)
+    end
+    return test
+  end
```

Comment:
Not working
```
> [nil, true, 99].any?
 => true 
> [nil, true, 99].my_any?
Traceback (most recent call last):
        7: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        6: from (irb):15
        5: from ennMethods.rb:44:in `my_any?'
        4: from ennMethods.rb:5:in `my_each'
        3: from ennMethods.rb:5:in `each'
        2: from ennMethods.rb:6:in `block in my_each'
        1: from ennMethods.rb:45:in `block in my_any?'
LocalJumpError (no block given (yield))
```

Reply:
Change made. Initially `my_any?` was made to receive always a block, with a syntax similar to this:
`[1,2,12,3].my_any? do |x|
  x.odd?
end`

Now it works for cases where no block is given (`[nil, true, 99].my_any?`) and for cases where a pattern is given (`["area", "all", "name"].my_any?(/a/)`)